File: src/main/java/com/bdev/hogwarts_api/rest_service/cabinet/CabinetRestService.java
package com.bdev.hogwarts_api.rest_service.cabinet;

import acropollis.municipali.security.common.dto.MunicipaliUserInfo;
import com.bdev.hogwarts_api.data.dto.cabinet.Cabinet;

import java.util.List;

public interface CabinetRestService {
    List<Cabinet> getAllCabinets(MunicipaliUserInfo userInfo);
    Cabinet getCabinet(MunicipaliUserInfo userInfo, long cabinetId);
    long saveCabinet(MunicipaliUserInfo userInfo, Cabinet cabinet);
    void editCabinet(MunicipaliUserInfo userInfo, Cabinet cabinet);
    void deleteCabinet(MunicipaliUserInfo userInfo, long cabinetId);
}

File: src/main/java/com/bdev/hogwarts_api/rest_service/cabinet/CabinetRestService.kt
package com.bdev.hogwarts_api.rest_service.cabinet

import acropollis.municipali.security.common.dto.MunicipaliUserInfo
import com.bdev.hogwarts_api.data.dto.cabinet.Cabinet

interface CabinetRestService {
    fun getAllCabinets(userInfo: MunicipaliUserInfo): List<Cabinet>
    fun getCabinet(userInfo: MunicipaliUserInfo, cabinetId: Long): Cabinet
    fun saveCabinet(userInfo: MunicipaliUserInfo, cabinet: Cabinet): Long
    fun editCabinet(userInfo: MunicipaliUserInfo, cabinet: Cabinet)
    fun deleteCabinet(userInfo: MunicipaliUserInfo, cabinetId: Long)
}
